ui/packages/catalog-ui-search/src/main/webapp/component/search-form/forms-sharing/search-form-sharing.collection.js

```diff
@@ -12,109 +12,112 @@
  * <http://www.gnu.org/licenses/lgpl.html>.
  *
  **/
- /*global require*/
- const _ = require('underscore');
- const $ = require('jquery');
- const Backbone = require('backbone');
- const SearchForm = require('../search-form');
- const Common = require('js/Common');
- const user = require('component/singletons/user-instance');
+/* global require */
+const _ = require('underscore')
+const $ = require('jquery')
+const Backbone = require('backbone')
+const SearchForm = require('../search-form')
+const Common = require('js/Common')
+const user = require('component/singletons/user-instance')
 
- let sharedTemplates = [];
- let promiseIsResolved = false;
- const sharedSearchFormPromise = () => $.ajax({
-    type: 'GET',
-    context: this,
-    url: './internal/forms/query',
-    contentType: 'application/json',
-    success: function (data) {
-        sharedTemplates = data;
-        promiseIsResolved = true;
-    }
-});
+let sharedTemplates = []
+let promiseIsResolved = false
+const sharedSearchFormPromise = () => $.ajax({
+  type: 'GET',
+  context: this,
+  url: './internal/forms/query',
+  contentType: 'application/json',
+  success: function (data) {
+    sharedTemplates = data
+    promiseIsResolved = true
```

Comment:
❗️ I might very well be misunderstanding something here, but this appears to be defeating the idea of reactive style that promises provide. If you're checking for promise resolution instead of performing an action based on success/failure we're back to polling.

Reply:
In all fairness, he didn't modify any of this code, it's diff is a result of running the formatter. However, @cantstoptheunk could probably do a little clean up here.

Reply:
Ahh. I took a stab at this...haven't actually tested to make sure it works. In case it helps:

```
const fetchSearchFormData = async () => {
    return fetch('./internal/forms/query', {
        headers: {
            'X-Requested-With': 'XMLHttpRequest',
            'Content-Type': 'application/json'
        }
    });
}

module.exports = Backbone.AssociatedModel.extend({
    model: SearchForm,
    defaults: {
        doneLoading: false,
        sharedSearchForms: []
    },
    initialize: function () {
        this.retrieveAndProcessFormData();
    },
    async retrieveAndProcessFormData() {
        let json;
        
        try {
            const response = await fetchSearchFormData();
            json = await response.json();
        } catch(error) {
            //TODO: Probably should let user know something's up
            console.error(error);
            return;
        }

        $.each(json, (index, value) => {
            if (!this.checkIfShareable(value)) {
                return;
            }

            const utcSeconds = value.created / 1000;
            const date = new Date(0);
            date.setUTCSeconds(utcSeconds);
            this.addSearchForm(new SearchForm({
                createdOn: Common.getHumanReadableDate(d),
                id: value.id,
                name: value.title,
                description: value.description,
                type: 'custom',
                filterTemplate: JSON.stringify(value.filterTemplate),
                accessIndividuals: value.accessIndividuals,
                accessGroups: value.accessGroups,
                createdBy: value.creator,
                owner: value.owner,
                querySettings: value.querySettings
            }));
        });
        this.doneLoading();
    },
    relations: [{
        type: Backbone.Many,
        key: 'sharedSearchForms',
        collectionType: Backbone.Collection.extend({
            model: SearchForm,
            initialize: function () {}
        })
    }],
    checkIfShareable: function (template) {
        if (this.checkIfInGroup(template) || this.checkIfInIndividiuals(template)) {
            return true;
        }
        return false;
    },
    checkIfInGroup: function (template) {
        let myGroups = user.get('user').get('roles');
        let roleIntersection = myGroups.filter(function (n) {
            return template.accessGroups.indexOf(n) !== -1;
        });

        return !_.isEmpty(roleIntersection);
    },
    checkIfInIndividiuals: function (template) {
        let myEmail = [user.get('user').get('email')];
        let accessIndividualIntersection = myEmail.filter(function (n) {
            return template.accessIndividuals.indexOf(n) !== -1;
        });

        return !_.isEmpty(accessIndividualIntersection);
    },
    addSearchForm: function (searchForm) {
        this.get('sharedSearchForms').add(searchForm);
    },
    getDoneLoading: function () {
        return this.get('doneLoading');
    },
    doneLoading: function () {
        this.set('doneLoading', true);
    },
    getCollection: function () {
        return this.get('sharedSearchForms');
    },
});
```